src/general/index.ts

```diff
@@ -226,3 +226,54 @@ export function downloadFile(downloadConfig: { file: File | null; type: string;
     console.error('File download failed - ', err);
   }
 }
+
+
+/**
+ * Returns the value at given path from the source object. If path is not found then default value is returned.
+ *
+ * @param { { [key: string]: unknown } } obj - Source object
+ * @param {string} path - Path to desired key inside source object
+ *
+ * @remarks
+ * Provide a default value always to avoid unexpected behavior
+ *
+ * @example
+ * ```
+ * const obj = { a: { b: [ 56, 75, 23 ], d: 1 }, e: 2 };
+ *
+ * getData(obj, 'a.d', null) // 1
+ * getData(obj, 'e', null) // 2
+ * getData(obj, 'a.d.e', 'random') // 'random'
+ * getData(obj, 'a.b[0]', null) // 56
+ * getData(obj, 'a.b.[2]', null) // 23
+ * ```
+ */
+export function getData(obj: { [key: string]: unknown }, path: string, def: null | unknown = null): { [key: string]: unknown } | null | unknown {
+
+  const sanitzePath = (currPath: string) => {
+
+    // 'a.[0].b.c' => 'a.0.b.c'
+
+    let sanitizedPath = String(currPath).replaceAll('[', '.').replaceAll(']', '.').replaceAll('..', '.');
+
+    const isLastIndexDot = sanitizedPath.lastIndexOf('.') === sanitizedPath.length - 1;
+
+    sanitizedPath = sanitizedPath.slice(0, isLastIndexDot ? sanitizedPath.lastIndexOf('.') : sanitizedPath.length);
+
+    return sanitizedPath;
+  };
+
+  try {
+    const newPathArray = String(sanitzePath(path)).split('.');
+
+    for (const path of newPathArray) {
+      obj = obj[path] as { [key: string]: unknown };
+    }
+
+    return obj ? obj : def;
+
+  } catch (e) {
+    console.error('Error while using getData', e);
+    throw e;
```

Comment:
in case of error, should we return default value?

Reply:
Makes sense.. Will fix